### PR TITLE
docs: .NET Agent: Add clarifying examples to SetTransactionName API doc

### PR DESCRIPTION
**Pull request by nr-ahemsath**
This PR updates the .NET agent API doc, specifically for the `SetTransactionName()` API call.  I mainly added expanded examples, one for a web app and one for a console app, that should help clarify what this API does and does not do.

### CODE REVIEW

File: src/content/docs/apm/agents/net-agent/net-agent-api/net-agent-api.mdx [hunk 1]
```diff
@@ -2334,7 +2334,7 @@ The following list contains the different calls you can make with the API, inclu
 
     ### Description
 
-    Sets the name of the current transaction. Before you use this call, ensure you understand the implications of [metric grouping issues](/docs/agents/manage-apm-agents/troubleshooting/metric-grouping-issues).
+    Sets the name (other than the initial `WebTransaction` or `OtherTransaction` base name) of the current transaction. Before you use this call, ensure you understand the implications of [metric grouping issues](/docs/agents/manage-apm-agents/troubleshooting/metric-grouping-issues).
```
Comment (tippmar-nr) [inline src/content/docs/apm/agents/net-agent/net-agent-api/net-agent-api.mdx:2337]: I feel like this needs to be re-worded a bit -- maybe to say something like "Sets a custom transaction name (appearing after the `WebTransaction` or `OtherTransaction` prefix) for the current transaction. ..." ?

The example code makes it very clear, but the words here don't.

Comment (nr-ahemsath) [reply src/content/docs/apm/agents/net-agent/net-agent-api/net-agent-api.mdx]: Thanks for the feedback.  I wasn't super happy with my initial wording, and I like yours better.  I pushed a change.

**Review by tippmar-nr — APPROVED** [review]
Looks good!

**Review by adutta-newrelic — APPROVED** [review]
Minor changes made to the document. approved from docs team

**Pull request merged**